src/libecalc/core/consumers/compressor/component.py

```diff
@@ -132,13 +151,30 @@ def evaluate(
                 timesteps=evaluated_timesteps,
                 unit=Unit.NONE,
             ),
-            outlet_pressure_before_choking=TimeSeriesFloat(
-                values=aggregated_result.outlet_pressure_before_choking
-                if aggregated_result.outlet_pressure_before_choking
-                else [np.nan for _ in evaluated_timesteps],
-                timesteps=evaluated_timesteps,
-                unit=Unit.BARA,
-            ),
+            outlet_pressure_before_choking=outlet_pressure_before_choke,
+            stages=[
+                Stage(
+                    name="inlet",
+                    stream_condition=StreamCondition(
+                        rate=total_requested_rate,
+                        pressure=operational_settings.inlet_pressure,
+                    ),
+                ),
+                Stage(
```

Comment:
Should be easy to extend this list with more stages from models.

Reply:
I like the start to more structured add the different stage/phases etc of the stream that you are adding though :)